### PR TITLE
Jj add authorization

### PR DESCRIPTION
**Pull request by jessjno**
limited routes using except and only, added authorization in controllers using before_action, hid links from unauthorized users, hid private users

### CODE REVIEW

File: app/controllers/comments_controller.rb
Comment (Testell) [inline app/controllers/comments_controller.rb:1]: nice job with the is_an_authorized_user function, it covers all current scenarios we need to check for

File: config/routes.rb
Comment (Testell) [inline config/routes.rb:1]: nice you secured all the routes we needed

**Review by Testell — APPROVED** [review]

File: app/controllers/follow_requests_controller.rb [hunk 1]
```diff
@@ -67,4 +68,10 @@ def set_follow_request
     def follow_request_params
       params.require(:follow_request).permit(:recipient_id, :sender_id, :status)
     end
+
+    def ensure_current_user_is_owner
```
Comment (hatcheta1) [inline app/controllers/follow_requests_controller.rb:72]: nice job securing destroy, update, and edit with ```ensure_current_user_is_owner```

File: app/views/comments/_comment.html.erb [hunk 2]
```diff
@@ -10,12 +10,14 @@
       <p><%= comment.body %></p>
     </div>
     <div>
-      <%= link_to edit_comment_path(comment), class: "btn btn-link btn-sm text-muted" do %>
+      <% if current_user == comment.author %>
```
Comment (hatcheta1) [inline app/views/comments/_comment.html.erb:13]: nice job limiting edit/delete functions to the comment author

**Review by hatcheta1 — APPROVED** [review]
looks good!

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class LikesController < ApplicationController
   before_action :set_like, only: %i[ show edit update destroy ]
+  before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
```
Comment (dgarcia13-99) [inline app/controllers/likes_controller.rb:3]: nice job adding authorization for likes! it takes care of limiting access to these actions

File: app/controllers/follow_requests_controller.rb (hunk 1 above)
Comment (dgarcia13-99) [reply app/controllers/follow_requests_controller.rb:72]: agree! Great way to ensure the current user is the only one able to perform these actions

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
+  before_action :is_an_authorized_user, only: [:destroy, :create]
```
Comment (marywebby) [inline app/controllers/comments_controller.rb:3]: great job implementing this before_action!

File: app/controllers/follow_requests_controller.rb [hunk 5]
```diff
@@ -1,5 +1,6 @@
 class FollowRequestsController < ApplicationController
   before_action :set_follow_request, only: %i[ show edit update destroy ]
+  before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
```
Comment (marywebby) [inline app/controllers/follow_requests_controller.rb:3]: great job on including all the applicable methods here

File: app/controllers/follow_requests_controller.rb (hunk 1 above)
Comment (marywebby) [inline app/controllers/follow_requests_controller.rb:72]: Looks like this method is covering both sender and recipient, could also make one for @follow_request.recipient, but doesn't look necessary right now

File: app/controllers/likes_controller.rb (hunk 3 above)
Comment (marywebby) [inline app/controllers/likes_controller.rb:3]: nice job implementing `Likes`!

File: app/controllers/likes_controller.rb [hunk 6]
```diff
@@ -66,4 +67,10 @@ def set_like
     def like_params
       params.require(:like).permit(:fan_id, :photo_id)
     end
+
+    def ensure_current_user_is_owner
+    if current_user != @like.fan
```
Comment (marywebby) [inline app/controllers/likes_controller.rb:72]: 👍

File: app/controllers/photos_controller.rb [hunk 7]
```diff
@@ -67,4 +71,10 @@ def set_photo
     def photo_params
       params.require(:photo).permit(:image, :comments_count, :likes_count, :caption, :owner_id)
     end
+
+    def ensure_current_user_is_owner
+      if current_user != @photo.owner
```
Comment (marywebby) [inline app/controllers/photos_controller.rb:76]: good job changing this to @photo.owner!

File: app/views/photos/_photo.html.erb [hunk 8]
```diff
@@ -7,13 +7,15 @@
     </h2>
 
     <div>
+      <% if current_user == photo.owner %>
```
Comment (marywebby) [inline app/views/photos/_photo.html.erb:10]: good limit to only photo owners!

File: app/views/users/show.html.erb [hunk 9]
```diff
@@ -4,6 +4,8 @@
   </div>
 </div>
 
+
+<% if current_user ==@user || !@user.private? || current_user.leaders.include?(@user) %>
```
Comment (marywebby) [inline app/views/users/show.html.erb:8]: make sure to add in a space between the `==` and `@user`, just to be consistent with syntax

**Review by marywebby — APPROVED** [review]
Great job on this!

**Review by dgarcia13-99 — APPROVED** [review]

**Pull request closed without merging**